consolidated s390 device configuration

steffen-maier opened the pull request:
Consolidate the persistent and dynamic configuration of s390-specific devices in Linux distributions by delegating the configuration to the existing framework `zdev` from s390-tools.

This pull request prepares consolidated s390 device configuration in `anaconda`.
The top commit [see its description] **depends** on a certain commit from https://github.com/ibm-s390-linux/s390-tools/pull/158 and thus https://github.com/ibm-s390-linux/s390-tools/releases/tag/v2.31.0.

Zdev's job is to perform low-level configuration after which the user gets architecture-independent objects such as block devices, or SCSI devices. Those can and should in turn be configured with existing common code mechanisms. So there's a clear separated layering for configuration duties.

In particular, the s390-specific devices currently are: DASD (traditional disk), and ZFCP (scsi). Zdev has a stable command line user interface and abstracts from sysfs and from a persistent configuration representation. Zdev encapsulates configuration details. Systems management code can simply delegate configuration to zdev and thus reduce architecture-specific code.

_This improves user experience, serviceability, maintainability, and reduces test effort._

@jstodola @poncovka @sharkcz

Even though this is a draft pull request, I would appreciate review comments.
It's only a draft until we sorted out the dependencies and merge order of pull requests for different related projects.

StorageGhoul commented:
Can one of the admins verify this patch?

vojtechtrefny commented:
Jenkins, ok to test.

steffen-maier commented:
This reminds me of a similar chicken and egg problem like https://github.com/rhinstaller/anaconda/pull/5249#issuecomment-1771168157.
**_How to untangle the dependencies?:_**

https://github.com/storaged-project/blivet/pull/1162/commits/2181a7e2b1b691783ab98b6fd41db613a55d02d1#diff-1aea9ad79758b521c7bab07267b2e2893723c4eddd2ea591e912742ec25e3be5L531
removes `DASDDevice.opts` here in blivet
but the **anaconda** unit tests (which I can hardly update here in a blivet PR ?) running here in the blivet git workflow now complain because they still expect old blivet code:

=================================== FAILURES ===================================
_________________ DASDInterfaceTestCase.test_find_formattable __________________
[unit_tests/pyanaconda_tests/modules/storage/test_module_dasd.py:82: ](https://github.com/rhinstaller/anaconda/blob/80aa6d46e5f71f111bdf88461920a5ca55e53e46/tests/unit_tests/pyanaconda_tests/modules/storage/test_module_device_tree.py#L168)
E           TypeError: DiskDevice.__init__() got an unexpected keyword argument 'opts'

____________ DeviceTreeInterfaceTestCase.test_get_dasd_device_data _____________
[unit_tests/pyanaconda_tests/modules/storage/test_module_device_tree.py:168: ](https://github.com/rhinstaller/anaconda/blob/80aa6d46e5f71f111bdf88461920a5ca55e53e46/tests/unit_tests/pyanaconda_tests/modules/storage/test_module_dasd.py#L82)
E           TypeError: DiskDevice.__init__() got an unexpected keyword argument 'opts'

vojtechtrefny reviewed (approved):
I am not an s390 expert, but this looks good to me in general, thank you.

vojtechtrefny commented:
You can ignore the *ci/x86_64/f_37* CI job failure, that's a problem in our CI configuration.

vojtechtrefny commented:
> This reminds me of a similar chicken and egg problem like [rhinstaller/anaconda#5249 (comment)](https://github.com/rhinstaller/anaconda/pull/5249#issuecomment-1771168157). **_How to untangle the dependencies?:_**

I also answered in the Anaconda PR, but short version is that we will merge the Blivet change even with the Anaconda unit tests failing here in our CI and then make sure the Anaconda PR works with the latest Blivet build and also later make sure to release Blivet and Anaconda together.

steffen-maier commented:
Just updated the commit descriptions with the actual upstream commits since https://github.com/ibm-s390-linux/s390-tools/pull/158 got merged.
Added explicit package dependency in spec file, to model the dependencies (similar to https://github.com/rhinstaller/anaconda/pull/5250#pullrequestreview-1682480291).
No code changes.

steffen-maier commented:
> Just updated the commit descriptions with the actual upstream commits since [ibm-s390-linux/s390-tools#158](https://github.com/ibm-s390-linux/s390-tools/pull/158) got merged. Added explicit package dependency in spec file, to model the dependencies (similar to [rhinstaller/anaconda#5250 (review)](https://github.com/rhinstaller/anaconda/pull/5250#pullrequestreview-1682480291)). No code changes.

Now the s390-tools upstream version v2.31.0 is known and I updated the commit touching the spec file ("DASDDevice: dracut_setup_args() without deprecated dasd.conf (#1802482,#1937049)").
While at it, replaced all references to commits on other packages with full URLs instead of just package name and short git commit ID (idea from https://github.com/dracutdevs/dracut/pull/2534#issuecomment-1933744932).

vojtechtrefny commented:
Replaced by #1259
